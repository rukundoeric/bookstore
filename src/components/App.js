import React from 'react';
import BooksList from './containers/BooksList';
import BooksForm from './containers/BooksForm';
import Header from './layouts/Header';
import '../assets/css/style.css';

function App() {
  return (
    <div className="content">
      <Header />
      <div className="container mt-3">
        <BooksList />
      </div>
      <div className="container mt-5">
        <BooksForm />
      </div>
    </div>
  );
}

export default App;
