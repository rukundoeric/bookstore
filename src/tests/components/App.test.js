import '@testing-library/jest-dom';
import { render, cleanup } from '@testing-library/react';
import App from '../../components/App';

afterEach(cleanup);

it('Should match snapshot', () => {
  const { asFragment } = render(<App />);
  expect(asFragment(<App />)).toMatchSnapshot();
});
