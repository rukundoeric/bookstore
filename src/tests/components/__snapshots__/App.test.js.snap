// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="content"
  >
    <header
      class="header"
    >
      <div
        class="container d-flex justify-content-between"
      >
        <div
          class="d-flex left justify-content-center align-items-center"
        >
          <h6>
            BookStore CMS
          </h6>
        </div>
        <div
          class="d-flex right justify-content-center align-items-center"
        >
          <span
            class="d-flex justify-content-center align-items-center"
          >
            <i>
              B
            </i>
          </span>
        </div>
      </div>
    </header>
    <div
      class="container mt-2"
    >
      <div
        class="p-3 mb-5"
      >
        <div>
          <div
            class="row"
          >
            <label
              class="form-label d-flex"
              for="category"
            >
              <select
                aria-label="Default select example"
                class="form-select"
                required=""
              >
                <option
                  value="All"
                >
                  All
                </option>
                <option
                  value="Fiction"
                >
                  Fiction
                </option>
                <option
                  value="Biography"
                >
                  Biography
                </option>
                <option
                  value="History"
                >
                  History
                </option>
                <option
                  value="Horror"
                >
                  Horror
                </option>
                <option
                  value="Kids"
                >
                  Kids
                </option>
                <option
                  value="Learning"
                >
                  Learning
                </option>
                <option
                  value="Sci-Fi"
                >
                  Sci-Fi
                </option>
              </select>
            </label>
          </div>
        </div>
        <div
          class="listItems"
        >
          <div
            class="mt-3 p-4 item bg-body rounded d-flex flex-column"
          >
            <span
              class="category"
            >
              kids
            </span>
            <span
              class="title"
            >
              Title
            </span>
            <span
              class="id"
            >
              SJSJ887NSKNKBK
            </span>
            <div
              class="pt-3"
            >
              <button
                class="delButton"
                data-id="SJSJ887NSKNKBK"
                type="button"
              >
                Delete
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="container line"
    />
    <div
      class="container mt-2"
    >
      <form>
        <div
          class="container bookform"
        >
          <span
            class="title"
          >
            ADD NEW BOOK
          </span>
          <div
            class="d-flex row mt-3"
          >
            <div
              class="col-lg-7"
            >
              <input
                class="form-control"
                id="title"
                name="title"
                placeholder="Title"
                required=""
                type="text"
                value=""
              />
            </div>
            <div
              class="col-lg-3"
            >
              <select
                aria-label="Default select example"
                class="form-select"
                name="category"
                required=""
              >
                <option
                  value="Fiction"
                >
                  Fiction
                </option>
                <option
                  value="Biography"
                >
                  Biography
                </option>
                <option
                  value="History"
                >
                  History
                </option>
                <option
                  value="Horror"
                >
                  Horror
                </option>
                <option
                  value="Kids"
                >
                  Kids
                </option>
                <option
                  value="Learning"
                >
                  Learning
                </option>
                <option
                  value="Sci-Fi"
                >
                  Sci-Fi
                </option>
              </select>
            </div>
            <div
              class="col-lg-2"
            >
              <button
                class="btn btn-primary px-5"
                type="submit"
              >
                ADD BOOK
              </button>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
